packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -769,6 +769,22 @@ contract VotingV2 is Staker, OracleInterface, OracleAncillaryInterface, OracleGo
         return (currentActiveRequests() && getVotePhase() == Phase.Reveal);
     }
 
+    // This function must be called before any tokens are staked. It updates the voter's pending stakes to reflect the new amount
+    // to stake. These updates are only made if we are in a reveal phase with active price requests. This is required in
+    // order to appropriately calculate a voter's trackers and avoid slashing them for amounts staked during an active reveal phase.
+    function _computePendingStakes(address voterAddress, uint256 amount) internal override {
+        if (_inActiveReveal()) {
+            uint256 currentRoundId = getCurrentRoundId();
+            // We now can freeze the round variables as we do not want the cumulativeActiveStakeAtRound to change based on the stakes
+            // that occur during the active reveal phase.
```

Comment:
perhaps add something that says _this only happens if the first action within the active reveal is someone staking, rather than someone revealing their vote_. this is true, right?

Reply:
In the scope of this function we run `_freezeRoundVariables` every time someone stakes during an active reveal phase. But given how `_freezeRoundVariables` works the variables are only frozen once, either with the first stake or reveal in the active reveal phase. 

_Since subsequent calls to _freezeRoundVariables only freeze variables the first time, variables in a round are only frozen once, either with the first stake or with the first reveal in an active reveal phase_